### PR TITLE
Wdt 414 k8s filter

### PR DESCRIPTION
**Pull request by jshum2479**
This add option for discovery to generate model files that is suitable for K8s deployment.  Also, there is a  new tool prepareModel that takes a list of model files instead of a domain home.

### CODE REVIEW

File: core/src/main/python/wlsdeploy/util/cla_utils.py [hunk 1]
```diff
@@ -973,19 +979,64 @@ def is_boolean_switch(self, key):
...
 
+    def is_target_switch(self, key):
+        return key == self.TARGET_SWITCH
+
+    def _validate_target_arg(self, value):
+        method_name = 'validate_kubernetes_script_file_switch'
+
+        # Check for allowed targets
+
+        allowed_values = [ 'k8s' ]
```
Comment (rakillen) [inline core/src/main/python/wlsdeploy/util/cla_utils.py:1003]: Should any value be allowed here, as long as the directory exists? Not sure if we are allowing users to extend this and make their own? Opinions?

File: core/src/main/python/wlsdeploy/tool/util/variable_injector.py [hunk 2]
```diff
@@ -2,29 +2,25 @@
 Copyright (c) 2018, 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (rakillen) [inline core/src/main/python/wlsdeploy/tool/util/variable_injector.py:2]: Update copyright

File: core/src/main/python/prepare_model.py [hunk 3]
```diff
@@ -0,0 +1,546 @@
+# Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+# ------------
+# Description:
+# ------------
+#
+#   This code prepare a list of models for k8s oeprator
```
Comment (rakillen) [inline core/src/main/python/prepare_model.py:8]: *operator

File: core/src/main/python/prepare_model.py [hunk 4]
```diff
@@ -0,0 +1,546 @@
...
+    """
+    Get message using the bundle.
+    :param key: bundle key
+    :param args: bundle arguments
+    :return:
+    """
+    return ExceptionHelper.getMessage(key, list(args))
+
+if __name__ == "__main__":
+    main()
```
Comment (rakillen) [inline core/src/main/python/prepare_model.py:544]: allow new Jython, add initial log statement:
```
if __name__ == '__main__' or __name__ == 'main':
    WebLogicDeployToolingVersion.logVersionInfo(_program_name)
    main(sys.argv)
```

File: core/src/main/python/wlsdeploy/tool/discover/k8s_operator_filter.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                        del servers[server][delthis]
+
+        if topology.has_key('SecurityConfiguration'):
+            for delthis in ['NodeManagerPasswordEncrypted', 'NodeManagerUsername' ]:
+                if topology['SecurityConfiguration'].has_key(delthis):
+                    del topology['SecurityConfiguration'][delthis]
+            if len(topology['SecurityConfiguration'].keys()) == 0:
+                del topology['SecurityConfiguration']
+
+        if topology.has_key('SeverTemplate'):
```
Comment (rakillen) [inline core/src/main/python/wlsdeploy/tool/discover/k8s_operator_filter.py:51]: *ServerTemplate

File: core/src/main/python/wlsdeploy/tool/discover/k8s_operator_filter.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2018, 2020, Oracle Corporation and/or its affiliates.
```
Comment (rakillen) [inline core/src/main/python/wlsdeploy/tool/discover/k8s_operator_filter.py:1]: new file,
```
# Copyright (c) 2020, Oracle Corporation and/or its affiliates.
```

File: core/src/main/python/wlsdeploy/tool/util/filter_helper.py [hunk 7]
```diff
@@ -30,20 +30,36 @@ def apply_filters(model, tool_type):
...
+
 
     try:
-        if os.path.isfile(__filter_file_location):
+        if model_context:
+            configuration = model_context.get_target_configuration()
+        filters_dictionary = {}
+        if configuration and 'model_filters' in configuration:
+            filters_dictionary = configuration['model_filters']
+            target_filter_path =  os.path.join(os.environ.get('WLSDEPLOY_HOME', ''), 'lib', 'target',
```
Comment (rakillen) [inline core/src/main/python/wlsdeploy/tool/util/filter_helper.py:42]: Can the directory be lib/targets, similar to lib/injectors and lib/typedefs?

File: core/src/main/python/wlsdeploy/util/cla_utils.py (hunk 1 above)
Comment (jshum2479) [reply core/src/main/python/wlsdeploy/util/cla_utils.py:1003]: Base on what we discussed on slack, we can allow different values

Comment (jshum2479) [reply core/src/main/python/wlsdeploy/util/cla_utils.py:1003]: done

File: core/src/main/python/wlsdeploy/tool/util/variable_injector.py (hunk 2 above)
Comment (jshum2479) [reply core/src/main/python/wlsdeploy/tool/util/variable_injector.py]: done

File: core/src/main/python/prepare_model.py (hunk 3 above)
Comment (jshum2479) [reply core/src/main/python/prepare_model.py]: done

File: core/src/main/python/prepare_model.py (hunk 4 above)
Comment (jshum2479) [reply core/src/main/python/prepare_model.py:547]: done

File: core/src/main/python/wlsdeploy/tool/discover/k8s_operator_filter.py (hunk 5 above)
Comment (jshum2479) [reply core/src/main/python/wlsdeploy/tool/discover/k8s_operator_filter.py]: done

File: core/src/main/python/wlsdeploy/tool/discover/k8s_operator_filter.py (hunk 6 above)
Comment (jshum2479) [reply core/src/main/python/wlsdeploy/tool/discover/k8s_operator_filter.py]: done

File: core/src/main/python/wlsdeploy/tool/util/filter_helper.py (hunk 7 above)
Comment (jshum2479) [reply core/src/main/python/wlsdeploy/tool/util/filter_helper.py]: done

File: core/src/main/python/wlsdeploy/tool/discover/k8s_operator_filter.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+#
+def filter_model(model):
+    __cleanup_topology(model)
+    __cleanup_resources(model)
+
+def __cleanup_resources(model):
+    if model and 'resources' in model:
+        resources = model['resources']
+
+    for delthis in [ 'PartitionWorkManager', 'Partition', 'ResourceGroup', 'ResourceGroupTemplate']:
```
Comment (CarolynRountree) [inline core/src/main/python/wlsdeploy/tool/discover/k8s_operator_filter.py:18]: what about virtual hosts, resource manager and resource management?

File: core/src/main/python/wlsdeploy/util/target_configuration_helper.py [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+        k8s_create_script_handle.close()
+
+def format_as_secret(prop_name):
+    """
+    Format as X.Y.A.B  This is the pattern of the name.
+    :param prop_name: property name
+    :return: formatted name
+    """
+    name_lower_tokens = prop_name.lower().split('.')
```
Comment (CarolynRountree) [inline core/src/main/python/wlsdeploy/util/target_configuration_helper.py:81]: I might have missed something in the code that calls this, but isn't this just an attribute name at this point? If so how can it be always be unique. The variable injector makes a unique name for the property.

File: core/src/main/targetconfigs/k8s/k8s.json [hunk 10]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (CarolynRountree) [inline core/src/main/targetconfigs/k8s/k8s.json:1]: Does this allow for user defined variable injectors?

File: core/src/main/python/wlsdeploy/util/target_configuration_helper.py (hunk 9 above)
Comment (jshum2479) [reply core/src/main/python/wlsdeploy/util/target_configuration_helper.py:81]: The caller passing an unique name - self.__format_variable_name(location, attribute). I will rename the variable.

File: core/src/main/python/wlsdeploy/tool/discover/k8s_operator_filter.py (hunk 8 above)
Comment (jshum2479) [reply core/src/main/python/wlsdeploy/tool/discover/k8s_operator_filter.py:18]: added

File: core/src/main/python/prepare_model.py (hunk 4 above)
Comment (rakillen) [reply core/src/main/python/prepare_model.py:547]: also needs 
```
if __name__ == '__main__' or __name__ == 'main':
```

**Pull request merged**